packages/core/src/api.ts

```diff
@@ -0,0 +1,105 @@
+import { DSNLike } from '@sentry/types';
+import { urlEncode } from '@sentry/utils/object';
+import { DSN } from './dsn';
+
+const SENTRY_API_VERSION = '7';
+
+/** Helper class to provide urls to different Sentry endpoints. */
+export class API {
+  /** The internally used DSN object. */
+  private dsnObject: DSN;
+  /** Create a new instance of API */
+  public constructor(public dsn: DSNLike) {
+    this.dsnObject = new DSN(dsn);
+  }
+
+  /** Returns the DSN object. */
+  public getDSN(): DSN {
+    return this.dsnObject;
+  }
+
+  /** Returns a string with auth headers in the url to the store endpoint. */
+  public getStoreEndpoint(urlEncodedHeader: boolean = false): string {
```

Comment:
Ughh... can't we just create a new method for this? I really don't like using `boolean` arguments to functions, as for people not using typescript, it provides no meaning before they dive into source code